pandas/tests/io/pytables/test_store.py

```diff
@@ -298,6 +298,28 @@ def test_keys(self, setup_path):
             assert set(store.keys()) == expected
             assert set(store) == expected
 
+    def test_non_pandas_keys(self, setup_path):
+        class Table1(tables.IsDescription):
+            value1 = tables.Float32Col()
+
+        class Table2(tables.IsDescription):
+            value2 = tables.Float32Col()
+
+        class Table3(tables.IsDescription):
+            value3 = tables.Float32Col()
+
+        with ensure_clean_path(setup_path) as path:
+            with tables.open_file(path, mode="w") as h5file:
+                group = h5file.create_group("/", "group")
+                h5file.create_table(group, "table1", Table1, "Table 1")
+                h5file.create_table(group, "table2", Table2, "Table 2")
+                h5file.create_table(group, "table3", Table3, "Table 3")
+            with HDFStore(path) as store:
+                assert len(store.keys(kind="tables")) == 3
+                expected = {"/group/table1", "/group/table2", "/group/table3"}
+                assert set(store.keys(kind="tables")) == expected
+                assert set(store) == set()
```

Comment:
Shouldn't that be
```python
assert set(store.keys()) == set()
```
?

Reply:
This form is equivalent, as the store implements the `__iter__` method which just returns an iterator with `keys()`
I will change it to be explicit.

Reply:
Thanks for the explanation, that makes sense indeed.